src/api/auth/auth.service.ts

```diff
@@ -1,22 +1,34 @@
-import { Injectable, Post } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
 import { FirebaseClientService } from '../../firebase/firebase-client.service';
 import { SignUpDto } from './dto/sign-up.dto';
 import { SignInDto } from './dto/sign-in.dto';
+import { FirebaseAdminService } from 'src/firebase/firebase-admin.service';
 
 @Injectable()
 export class AuthService {
-  constructor(private readonly clientService: FirebaseClientService) {}
+  constructor(
+    private readonly firebaseClientService: FirebaseClientService,
+    private readonly firebaseAdminService: FirebaseAdminService,
+  ) {}
 
-  @Post()
   async singUp(signUpDto: SignUpDto) {
     const { email, password } = signUpDto;
-    const user = await this.clientService.createFirebaseUser(email, password);
-    return { email: user.email, emailVerified: user.emailVerified };
+    const user = await this.firebaseClientService.createFirebaseUser(
+      email,
+      password,
+    );
+    return user;
   }
 
-  @Post()
   async singIn(signInDto: SignInDto) {
     const { email, password } = signInDto;
-    return this.clientService.signInAndSendVerification(email, password);
+    return this.firebaseClientService.signInAndSendVerification(
+      email,
+      password,
+    );
+  }
+
+  async refreshToken(refreshToken: string) {
```

Comment:
did frontend send refresh token or hi send access token ?

Reply:
Frontend have to send only refresh token